cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/scorecompiler/ScoreCompilerImpl.java

```diff
@@ -65,6 +69,8 @@ public CompilationArtifact compile(Executable executable, Set<Executable> path)
             //than we match the references to the actual dependencies
             filteredDependencies = dependenciesHelper.matchReferences(executable, availableExecutables);
 
+            handleOnFailureCustomResults(availableExecutables);
```

Comment:
I think we can use here filteredDependencies and that way we don't have to recalculate it in `handleOnFailureCustomResults` each time - WDYT?